src/test/java/org/example/bookshop/controller/CategoryControllerTest.java

```diff
@@ -0,0 +1,319 @@
+            executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD)
+    @WithMockUser(username = "admin", authorities = {"ADMIN", "USER"})
+    @DisplayName("""
+            Should update the existing Category
+            """)
+    void updateBook_WithCorrectCreateCategoryRequestDto_Success() throws Exception {
+        //Given
+        UpdateCategoryRequestDto updateCategoryRequestDto
+                = new UpdateCategoryRequestDto();
+        updateCategoryRequestDto.setName("lol info new");
+        updateCategoryRequestDto.setDescription("some description new");
+
+        CategoryDto expected = new CategoryDto();
+        expected.setId(2L);
+        expected.setName("lol info new");
+        expected.setDescription("some description new");
+
+        String jsonRequest = objectMapper
+                .writeValueAsString(updateCategoryRequestDto);
+
+        //When
+        final MvcResult mvcResult = mockMvc
+                .perform(
+                        put("/categories/2")
+                                .content(jsonRequest)
+                                .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andReturn();
+
+        final MvcResult mvcResultGet = mockMvc.perform(
+                        get("/categories/2")
+                )
+                .andExpect(status().isOk())
+                .andReturn();
+
+        //Then
+        CategoryDto actual = objectMapper.readValue(
+                mvcResult.getResponse().getContentAsString(),
+                CategoryDto.class);
+        Assertions.assertNotNull(actual);
+        Assertions.assertNotNull(actual.getId());
+        assertTrue(EqualsBuilder.reflectionEquals(expected, actual));
+
+        CategoryDto actualGet = objectMapper.readValue(
+                mvcResultGet.getResponse().getContentAsString(),
+                CategoryDto.class);
+        Assertions.assertNotNull(actualGet);
+        Assertions.assertNotNull(actualGet.getId());
+        assertTrue(EqualsBuilder.reflectionEquals(expected, actualGet));
+    }
+
+    @Test
+    @Sql(scripts = "classpath:categories/add-categories.sql",
+            executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD)
+    @WithMockUser(username = "admin", authorities = {"ADMIN", "USER"})
+    @DisplayName("""
+            Should delete Category in DB and return code 204
+            """)
+    void deleteBook_BookExist_Success() throws Exception {
+        //When
+        mockMvc.perform(
+                        get("/categories/1")
+                )
+                .andExpect(status().isOk())
+                .andReturn();
+
+        MvcResult mvcResult = mockMvc.perform(
+                delete("/categories/1")
+        ).andExpect(status().isNoContent()).andReturn();
+
+        mockMvc.perform(
+                        get("/categories/1")
+                )
+                .andExpect(status().isNotFound())
```

Comment:
the same. Don’t use other endpoints to test results. your test cases should be isolated